views/js/previewer/proxy/test.js

```diff
@@ -62,17 +62,39 @@ define([
         review: 6
     });
     /**
-     * Finds ids of testPart, section and item in testMap for a given item position
+     * Updates testContext with item data
      * @param {Object} testMap
-     * @param {Number} position item position
-     * @returns {Object} object containing testPartId, sectionId, itemIdentifier
+     * @param {Number} position - item position
+     * @param {Object} testContext
+     * @param {Array} presetMap
      */
-    function findIds(testMap, position) {
-        const item = mapHelper.getJump(testMap, position);
+    function updateTestContextWithItem(testMap, position, testContext, presetMap) {
+        const jump = mapHelper.getJump(testMap, position);
+        const item = mapHelper.getItemAt(testMap, position);
         if (item) {
```

Comment:
you can use [early returns ](https://dev.to/yvonnickfrin/devtips-use-early-returns-to-avoid-nested-conditions-4i51)here.

Reply:
fixed